### PR TITLE
Hadoop cleanup

### PR DESCRIPTION
**Pull request by electrum**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsInputFile.java [hunk 1]
```diff
@@ -117,7 +119,10 @@ private FSDataInputStream openFile()
             }
             catch (IOException e) {
                 openFileCallStat.recordException(e);
-                throw e;
+                if (e instanceof FileNotFoundException) {
```
Comment (dain) [inline lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsInputFile.java:123]: With this form you are recording the exception in the stat object.  Is that what you want?  If not, I'd move this to a separate catch clause

File: lib/trino-hdfs/src/test/java/io/trino/filesystem/hdfs/TestHdfsFileSystemLocal.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    @Override
+    protected TrinoFileSystem getFileSystem()
+    {
+        return fileSystem;
+    }
+
+    @Override
+    protected Location getRootLocation()
+    {
+        return Location.of("viewfs://abc/");
```
Comment (dain) [inline lib/trino-hdfs/src/test/java/io/trino/filesystem/hdfs/TestHdfsFileSystemLocal.java:115]: Should we name this test `TestHdfsFileSystemViewFs`?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java [hunk 3]
```diff
@@ -540,11 +547,35 @@ private List<String> doListAllTables(String databaseName)
...
 
-        Path databaseMetadataDirectory = getDatabaseMetadataDirectory(databaseName);
-        List<String> tables = getChildSchemaDirectories(TABLE, databaseMetadataDirectory).stream()
-                .map(Path::getName)
-                .collect(toImmutableList());
-        return tables;
+        Location metadataDirectory = getDatabaseMetadataDirectory(databaseName);
+        try {
+            String prefix = metadataDirectory.toString();
+            Set<String> databases = new HashSet<>();
```
Comment (dain) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java:553]: should this be named `tables`?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java [hunk 4]
```diff
@@ -816,11 +842,12 @@ public synchronized void addPartitions(String databaseName, String tableName, Li
...
-                for (Path createdFile : createdFiles) {
-                    try {
-                        metadataFileSystem.delete(createdFile, false);
-                    }
-                    catch (IOException ignored) {
+                try {
+                    fileSystem.deleteFiles(createdFiles);
+                }
+                catch (IOException ex) {
+                    if (e.equals(ex)) {
```
Comment (dain) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java:849]: I think you want not equals

**Review by dain — APPROVED** [review]

File: lib/trino-hdfs/src/test/java/io/trino/filesystem/hdfs/TestHdfsFileSystemLocal.java (hunk 2 above)
Comment (electrum) [reply lib/trino-hdfs/src/test/java/io/trino/filesystem/hdfs/TestHdfsFileSystemLocal.java:115]: This is testing the local file system, using ViewFS to create a sandbox. The important behavior is that the operations are happening against the local file system. ViewFS shouldn't affect that.

File: lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsInputFile.java (hunk 1 above)
Comment (electrum) [reply lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsInputFile.java:123]: Yeah, the stat object actually has special handling for `FileNotFoundException`

**Pull request merged**